protocol/response.go

```diff
@@ -0,0 +1,14 @@
+package protocol
+
+// Response is a general purpose struct for capturing the response to a relay, received from an endpoint.
+// TODO_FUTURE: It only supports HTTP responses for now.
+type Response struct {
+	// Bytes is the response to a relay received from an endpoint.
+	// This can be a response to any type of RPC(GRPC, HTTP, etc.)
+	Bytes []byte
```

Comment:
No: the `Response` struct catpures the general response to a relay (please refer to the godoc on `Response`). As of now, it only supports HTTP, but this could change in the future: e.g. changing `Response` to an interface with HTTP, Websockets, etc. implementations.